in_toto/gpg/common.py

```diff
@@ -177,10 +182,20 @@ def parse_pubkey_bundle(data, keyid):
         sub_public_key = in_toto.gpg.common.parse_pubkey_payload(payload)
         sub_public_keys[sub_public_key["keyid"]] = sub_public_key
 
+      elif _type == PACKET_TYPES["signature_packet"]:
+        signature = in_toto.gpg.common.parse_signature_packet(data[packet_start:])
+        expires_after = signature["key_expire_time"]
+
     # The data might contain non-supported subkeys, which we just ignore
     except (ValueError, PacketVersionNotSupportedError,
         SignatureAlgorithmNotSupportedError):
       pass
+      
+    created = master_public_key["creation_date"]
+    if expires_after and created:
+      current_version_expiration = int(created) + int(expires_after)
+      if current_version_expiration > expiration_datetime:
+        expiration_datetime = current_version_expiration
```

Comment:
Although I agree it would simpler to derive the key expiration date from key creation time and key expiration time in verification, I did it this way to handle public keys with self-signatures that constantly update their expiration dates (e.g. keys which have their expiration dates extended once every year). This implementation is meant to ensure that a key's expiration date is always derived with the expiration time of its most recent self-signature packet.